### DISCUSSION TITLE
Import all modules

### DISCUSSION THREAD

**Discussion by mouhebsa** [primefaces/community#1378] [PrimeNG] [answered]
I need a way to recover the list of all modules so i can display it
and the end user can check the component that he needs to use

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

You can take it from all demos. Here is one: https://stackblitz.com/edit/vddt4q?file=src%2Fapp%2Fapp.module.ts

**Reply by mouhebsa** [reply]
actually i need to work with primeng library not extern library
is there a way to retrieve all graphical libraries list from primeng or even a webservice

**Comment by mouhebsa** [comment]
Hi,

Actually my app is a generator of application
I am displaying all primeng graphical component and user with drag drop can
work with any component.
Just i don’t want to work with hard coded component
I need a way to retrieve the list from primeng library or even is there a
webservice to retrieve the list of primeng components so i can display it

**Comment by khalilkhaled1** [comment]
Hi, we're having the same issue, we want to get the list of primeng available UI components available in a specific version dynamically (using a webservice preferably).

We want to get the UI components with their descriptions, inputs and outputs.

Is there a way to do that?
